modules/prometheus-source/pkg/prom/metricsquerier.go

```diff
@@ -1484,6 +1484,294 @@ func (pds *PrometheusMetricsQuerier) QueryReplicaSetsWithRollout(start, end time
 	return source.NewFuture(source.DecodeReplicaSetsWithRolloutResult, ctx.QueryAtTime(queryReplicaSetsWithRolloutOwner, end))
 }
 
+func (pds *PrometheusMetricsQuerier) QueryResourceQuotaSpecCPURequestAverage(start, end time.Time) *source.Future[source.ResourceQuotaSpecCPURequestAvgResult] {
```

Comment:
I guess I'm still a little bit confused about the Prometheus side of things... do the other changes in this package imply that we are not just scraping ResourceQuota data directly from the Kubernetes API (i.e. Cluster Cache), but also EMITTING it to a `/metrics` endpoint that a Prometheus could then scrape and ingest for the PrometheusMetricsQuerier to query?